client/src/plugins/error-tracking/ErrorTracking.js

```diff
@@ -0,0 +1,187 @@
+    const { config, _getGlobal } = this.props;
+
+    const editorID = await config.get(EDITOR_ID_CONFIG_KEY) || NON_EXISTENT_EDITOR_ID;
+
+    const releaseTag = Metadata.data.version;
+
+    try {
+
+      // We need to send a message to backend to initialize Sentry in the main process.
+      // Note that Sentry in renderer process (browser JS) and main process (node)
+      // has to be handled separately -> they have separate source maps and their Sentry
+      // modules are different (sentry-browser and sentry-node).
+      const backend = _getGlobal('backend');
+      backend.send('sentry:initialize', { dsn: this.SENTRY_DSN, releaseTag, editorID });
+
+      // Source map uploaded to Sentry from WebPack is tagged with the
+      // version number in package.json which is supposed to be the same
+      // with Metadata.data.version (except for dev environments
+      // - we don't initialize dev in Sentry.)
+      this._sentry.init({
+        dsn: this.SENTRY_DSN,
+        release: releaseTag
+      });
+
+      // OS information already exists by default in Sentry.
+      // We'll set editor ID and Camunda Modeler version.
+      this._sentry.configureScope(scope => { scope.setTag('editor-id', editorID); });
+
+      const { subscribe } = this.props;
+
+      // Send handled errors to Sentry.
+      subscribe('app.error-handled', (error) => {
+        if (!this._isInitialized) {
+          return;
+        }
+
+        this._sentry.captureException(error);
+      });
+
+      log('Initialized');
+
+      this._isInitialized = true;
+
+    } catch (err) {
+
+      log('Cannot initialize: ', err);
+    }
+  }
+
+  closeSentry() {
+
+    // tell backend to close Sentry-node instance.
+    const { _getGlobal } = this.props;
+    const backend = _getGlobal('backend');
+    backend.send('sentry:close');
+
+    this._sentry.close();
+    this._isInitialized = false;
+
+    log('Closed Sentry.');
+  }
+
+  async canInitializeSentry() {
+
+    if (!this.SENTRY_DSN) {
+      return {
+        result: false,
+        msg: 'No DSN set.'
+      };
+    }
+
+    const isCrashReportsEnabled = await this.isCrashReportsEnabled();
+
+    if (!isCrashReportsEnabled) {
+
+      return {
+        result: false,
+        msg: 'Crash reports are not enabled.'
+      };
+    }
+
+    return { result: true };
+  }
+
+  async isCrashReportsEnabled() {
+    const { config } = this.props;
+
+    const privacyPreferences = await config.get(PRIVACY_PREFERENCES_CONFIG_KEY);
+
+    return !!(privacyPreferences && privacyPreferences[CRASH_REPORTS_CONFIG_KEY]);
+  }
+
+  async recheckSentry() {
+    const { result } = await this.canInitializeSentry();
+
+    if (result !== this._isInitialized) {
+
+      // Status has changed:
+      // The user turned on / off Error Tracking option through
+      // Privacy Preferences modal.
+
+      if (result) {
+        this.initializeSentry();
+      } else {
+        this.closeSentry();
+      }
+    }
+  }
+
+  scheduleCheck() {
```

Comment:
Alternative: Emit an event or force reload of the client app. Not sure of any of these is viable.

Reply:
Also: If we go this route, what happens, if the user disables sentry integration after it got enabled?

Reply:
If the user unselects Error Tracking via Privacy Prefernces, we close Sentry instances on Frontend and Backend -> No more error tracking until the user selects the option again.

Since it'd be the PrivacyPreferences plugin that would emit this event, which infrastructure would you prefer us to use here in order to emit an event. We still don't have messaging functionality between our plugins: We can implement one with this PR, or we emit an event anyway instead of scheduling a check but that event would also be emitted to any client-side code. Any opinions here?

Reply:
> I think we should close Sentry as soon as error tracking is disabled. We are technically capable of doing this.

@barmac Let's discuss this here as this is the same topic. What do you sugges?